src/types.ts

```diff
@@ -0,0 +1,286 @@
+    finalized: boolean;
+
+    transactionCount: bigint;
+    transactionsSize: bigint;
+    transactionEnergyCost: bigint;
+}
+
+export interface ConsensusStatus {
+    bestBlock: string;
+    genesisBlock: string;
+    lastFinalizedBlock: string;
+
+    epochDuration: bigint;
+    slotDuration: bigint;
+    bestBlockHeight: bigint;
+    lastFinalizedBlockHeight: bigint;
+
+    finalizationCount: bigint;
+    blocksVerifiedCount: bigint;
+    blocksReceivedCount: bigint;
+
+    blockArriveLatencyEMA: number;
+    blockArriveLatencyEMSD: number;
+
+    blockReceiveLatencyEMA: number;
+    blockReceiveLatencyEMSD: number;
+
+    transactionsPerBlockEMA: number;
+    transactionsPerBlockEMSD: number;
+
+    blockReceivePeriodEMA?: number;
+    blockReceivePeriodEMSD?: number;
+
+    blockArrivePeriodEMA?: number;
+    blockArrivePeriodEMSD?: number;
+
+    finalizationPeriodEMA?: number;
+    finalizationPeriodEMSD?: number;
+
+    genesisTime: Date;
+    blockLastReceivedTime?: Date;
+    blockLastArrivedTime?: Date;
+    lastFinalizedTime?: Date;
+}
+
+export interface NextAccountNonce {
+    nonce: bigint;
+    allFinal: boolean;
+}
+
+export interface ReleaseSchedule {
+    timestamp: Date;
+    amount: bigint;
+    transactions: any;
+}
+
+export interface AccountReleaseSchedule {
+    total: bigint;
+    schedule: ReleaseSchedule[];
+}
+
+export interface AccountEncryptedAmount {
+    selfAmount: string;
+    startIndex: bigint;
+    incomingAmounts: string[];
+    numAggregated: number;
+}
+
+export interface VerifyKey {
+    schemeId: string;
+    verifyKey: string;
+}
+
+export interface CredentialPublicKeys {
+    keys: Record<number, VerifyKey>;
+    threshold: number;
+}
+
+export interface ChainArData {
+    encIdCredPubShare: string;
+}
+
+export interface Policy {
+    validTo: string; // "YYYYMM"
+    createdAt: string; // "YYYYMM"
+    revealedAttributes: Record<AttributeKey, string>;
+}
+
+interface SharedCredentialDeploymentValues {
+    ipIdentity: number;
+    credentialPublicKeys: CredentialPublicKeys;
+    policy: Policy;
+}
+
+export interface CredentialDeploymentValues
+    extends SharedCredentialDeploymentValues {
+    credId: string;
+    revocationThreshold: number;
+    arData: Record<string, ChainArData>;
+    commitments: CredentialDeploymentCommitments;
+}
+
+export interface InitialCredentialDeploymentValues
+    extends SharedCredentialDeploymentValues {
+    regId: string;
+}
+
+export interface AccountCredential {
+    type: 'normal' | 'initial';
+}
+
+export interface CredentialDeploymentCommitments {
+    cmmPrf: string;
+    cmmCredCounter: string;
+    cmmIdCredSecSharingCoeff: string[];
+    cmmAttributes: Record<AttributeKey, string>;
+    cmmMaxAccounts: string;
+}
+
+export interface NormalAccountCredential extends AccountCredential {
+    contents: CredentialDeploymentValues;
+}
+
+export interface InitialAccountCredential extends AccountCredential {
+    contents: InitialCredentialDeploymentValues;
+}
```

Comment:
These types could also be more precise, since `contents` and `type` are tied together

Reply:
Do you mean something like:
```
export interface InitialAccountCredential {
    type: 'initial';
    contents: InitialCredentialDeploymentValues;
}
```
instead?

Reply:
Yes, with this type I can check the `type` field and typescript would know the type of the `contents` as well